data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepository.java

```diff
@@ -32,6 +33,8 @@ Environment createEnvironmentAndEnvironmentRevision(
   Environment getEnvironment(EnvironmentId environmentId)
       throws ResourceNotFoundException, InternalServiceException;
 
+  List<Environment> listEnvironments(Cluster cluster) throws InternalServiceException;
```

Comment:
What do you mean by "cluster not found"? If there is no record associated with this cluster and account in DDB, this api just return an empty list.

Reply:
I think they are two different things. 

- Like ResouceNotFound for cluster, which means no such clsuter defined. 
- Emtpy list means there is no environment in this cluster, but cluster does exist.

My idea is following the same rule of list api, say list-services:
```
 aws ecs list-services --cluster not-exist

An error occurred (ClusterNotFoundException) when calling the ListServices operation: Cluster not found.
```

Reply:
The implementation would be different as we don't track the cluster.  Would be a front-end service validation thing to validate the cluster with ECS before invoke data service.

Reply:
I agree we need validation for non-exist cluster but I don't think we should take care of it at this level. I'm leaning to add it in data-service api rather than this DDB interacting level.

Reply:
Make sense. The cluster validation is not on DDB level.